src/VisualStudio/Core/Def/Remote/VisualStudioRemoteHostClientProvider.cs

```diff
@@ -56,18 +61,27 @@ public IWorkspaceService CreateService(HostWorkspaceServices workspaceServices)
                 // We don't want to bring up the OOP process in a VS cloud environment client instance
                 // Avoids proffering brokered services on the client instance.
                 if (!_globalOptions.GetOption(RemoteHostOptionsStorage.OOP64Bit) ||
-                    workspaceServices.Workspace is not VisualStudioWorkspace ||
+                    workspaceServices.Workspace is not (VisualStudioWorkspace or PreviewWorkspace) ||
                     workspaceServices.GetRequiredService<IWorkspaceContextService>().IsCloudEnvironmentClient())
                 {
                     // Run code in the current process
                     return new DefaultRemoteHostClientProvider();
                 }
 
-                return new VisualStudioRemoteHostClientProvider(workspaceServices.SolutionServices, _globalOptions, _vsServiceProvider, _threadingContext, _listenerProvider, _callbackDispatchers);
+                lock (_gate)
```

Comment:
>_gate [](http://example.com/codeflow?start=22&length=5)

This wouldn't work for MiscFiles or interactive workspaces, right?

Reply:
Correct. I'm not touching those yet. Misc is more concerning as it would constantly fork the oop workspace from a full solution to and empty over. I also don't understand interactive well enough to make that change yet.

Reply:
There will need to be multiple OOP workspaces.

Reply:
Sounds good. We can look into trialing this in 17.7 :-)